Dockerfile

```diff
@@ -1,21 +1,27 @@
+# Stage 1: Build the application
 FROM node:22-alpine AS builder
 
 WORKDIR /usr/src/app
 
+# Copy and install only production dependencies
 COPY package.json package-lock.json ./
+RUN npm ci --production
 
-RUN npm ci
-
+# Copy source files and build the project
 COPY . .
-
 RUN npm run build
 
-# Use a lightweight web server to serve the production build
+# Stage 2: Prepare the production image
 FROM nginx:stable-alpine AS production
 
+# Remove unnecessary files to reduce image size
+RUN rm -rf /var/cache/apk/* /tmp/*
```

Comment:
I would suggest removing this cleanup step, as it doesn’t provide any real benefit in a multi-stage build where intermediate layers aren’t included in the final image. The --production flag, however, is a useful addition to install only production dependencies.

Great job!

Reply:
Thank you for the thoughtful feedback, @origranot! You make an excellent point regarding the cleanup step. Since intermediate layers from the build stage don’t carry over to the final image, removing those files indeed doesn’t provide any real benefit.
I'm glad the addition of the --production flag is useful for minimizing the image size. Thanks again for the insightful review and the encouragement!